### PR TITLE
Ltac2 use hashmaps instead of ordered maps in suspected hot paths

### PR DESCRIPTION
**Pull request by SkySkimmer**
cf https://coq.zulipchat.com/#narrow/channel/237656-Coq-devs-.26-plugin-devs/topic/Some.20performance.20numbers.20for.208.2E19.20-.3E.208.2E20.2C.208.2E19.20-.3E.20master/near/479472477

Overlays:
- https://github.com/mit-plv/rewriter/pull/163

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by JasonGross** [comment]
coq-rewriter needs an overlay.  Is the bench compatible with overlays somehow?

File: plugins/ltac2/tac2env.ml [hunk 1]
```diff
@@ -134,16 +134,24 @@ let define_notation kn tac =
 
 let interp_notation kn = KNmap.find kn !ltac_notations
 
+let ml_tactic_name ~plugin tactic = {
```
Comment (JasonGross) [inline plugins/ltac2/tac2env.ml:137]: If you want to merge a version of this that just adds the wrapper without changing behavior, then we can get a rewiter overlay merged and this change can be benched against the Ltac2 code in the rewriter and Fiat Crypto

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by JasonGross** [comment]
│  21.3230   20.8210  -0.5020   -2.35%    40  coq-fiat-crypto-with-bedrock/src/PushButtonSynthesis/SolinasReductionReificationCache.v.html     │
│  60.9520   60.6950  -0.2570   -0.42%    27  coq-fiat-crypto-with-bedrock/src/Rewriter/Passes/ToFancyWithCasts.v.html                         │
│   7.3780    7.2270  -0.1510   -2.05%    35  coq-fiat-crypto-with-bedrock/src/PushButtonSynthesis/SolinasReductionReificationCache.v.html     │
│   3.3690    3.2530  -0.1160   -3.44%   344  coq-rewriter/src/Rewriter/Rewriter/Examples/PerfTesting/Plus0Tree.v.html                         │
│  39.2010   39.0850  -0.1160   -0.30%   236  coq-rewriter/src/Rewriter/Rewriter/Examples/PerfTesting/LiftLetsMap.v.html                       │

These might be real or noise, but the rest look like almost certainly noise

**Comment by Janno** [comment]
Just to make sure these results do not get lost on zulip[^1], here's the performance (in billion cpu instructions)  on the BlueRock.io code base. I excluded generated .v files that cannot possibly benefit from the change:

| Relative | Master   | MR       | Change 
|---------:|---------:|---------:|---------:|
|   -1.24% |  83723.3 |  82688.0 |  -1035.3


[^1]: https://coq.zulipchat.com/#narrow/channel/237656-Coq-devs-.26-plugin-devs/topic/Some.20performance.20numbers.20for.208.2E19.20-.3E.208.2E20.2C.208.2E19.20-.3E.20master/near/479496302

**Comment by ppedrot** [comment]
This kind of PR cannot hurt, but I think that it's missing the root of the problem. If the Ltac2 runtime is a bottleneck, it should be replaced by @SkySkimmer 's  [native compiler](https://github.com/SkySkimmer/coq-ltac2-compiler) or possibly some other middle-ground like e.g. leveraging the Coq VM as a runtime.

**Pull request closed without merging**